File: python/src/wdl_model/model/expressions/wdl_variable.py
from __future__ import annotations

from dataclasses import dataclass

from .wdl_expression import ComponentType, WdlExpression


@dataclass
class WdlVariable(WdlExpression):
    name: str | None = None

    def getName(self) -> str | None:
        return self.name

    def setName(self, name: str) -> None:
        self.name = name

    def __str__(self) -> str:
        return self.name or ""

    def componentType(self) -> ComponentType:
        return ComponentType.VARIABLE

File: typescript/src/model/expressions/wdl-variable.ts
/** Variable reference expression nodes. */
import { WdlExpressionComponentType, type WdlExpression } from './wdl-expression.js';

/** Models a variable reference such as `x` or `sample_name`. */
export class WdlVariable implements WdlExpression {
  /** Creates a variable reference from its source-level name. */
  public constructor(private nameValue?: string) {}

  /** Returns the referenced variable name. */
  public getName(): string | undefined {
    return this.nameValue;
  }

  /** Sets the referenced variable name. */
  public setName(name: string | undefined): void {
    this.nameValue = name;
  }

  /** Returns the broad expression family for this node. */
  public componentType(): WdlExpressionComponentType {
    return WdlExpressionComponentType.VARIABLE;
  }
}
